### PR TITLE
Support datasource permissions

### PR DESCRIPTION
**Pull request by julienduchesne**
Heavily inspired by dashboard permissions
Needed for https://github.com/grafana/terraform-provider-grafana/issues/205

### CODE REVIEW

File: datasource_permissions.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+type DatasourcePermission struct {
+	ID           int64  `json:"id"`
+	DatasourceID int64  `json:"datasourceId"`
+	UserID       int64  `json:"userId"`
+	UserEmail    string `json:"userEmail"`
+	TeamID       int64  `json:"teamId"`
+
+	// Permission levels are
+	// 1 = Query
+	Permission     int64  `json:"permission"`
```
Comment (inkel) [inline datasource_permissions.go]: Any reason why not an `int`? And similar to all the `int64` above, but this one in particular seems a bit overkill 😸

File: datasource_permissions.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	PermissionName string `json:"permissionName"`
+}
+
+type DatasourcePermissionsResponse struct {
+	DatasourceID int64 `json:"datasourceId"`
+	Enabled      bool  `json:"enabled"`
+	Permissions  []*DatasourcePermission
```
Comment (inkel) [inline datasource_permissions.go]: Is this one missing the JSON mapping?

File: datasource_permissions.go (hunk 1 above)
Comment (inkel) [reply datasource_permissions.go]: What do you think about adding an enum for this:

```go
type DatasourcePermission int

const (
	_ DatasourcePermission = iota // 0 is not a valid permission
	DatasourcePermissionQuery
)
```

And then define the field as `Permission DatasourcePermission`?

File: datasource_permissions.go (hunk 2 above)
Comment (julienduchesne) [reply datasource_permissions.go]: oops. It was auto-mapped due to being public I guess

**Review by inkel — APPROVED** [review]
:shipit: 🚀

**Pull request merged**